draft-ietf-masque-connect-udp-listen.md

```diff
@@ -164,6 +166,20 @@ and therefore is to be ignored). This document does not define any parameters
 for the connect-udp-listen header field value, but future documents might define
 parameters. Receivers MUST ignore unknown parameters.
 
+
+# The proxy-public-address Response Header Field {#response}
+
+Upon accepting the request, the proxy MUST allocate either one or both of IPv4address to port and IPv6address to port pairings. IPv4address, IPv6address and port are defined in {{!URI=RFC3986}}
+
+The proxy must then respond with the allocated IP and port pairings using the "proxy-public-address" response header defined as follows:
+
+~~~ ascii-art
+proxy-public-address = IP-port-pairing (, IP-port-pairing)
+IP-port-pairing = IPv4address:port / IPv6address:port
```

Comment:
This encoding is tricky because IPv6 addresses have colons in them. I would suggest doing something closer to RFC3986 host. Also, the ABNF format from that RFC requires double quotes to encode strings like colons.

```suggestion
IP-port-pairing = ( IPv4address / "[" IPv6address "]" ) ":" port
```